source/common/http/header_map_impl.h

```diff
@@ -52,6 +52,9 @@ public:
  */
 class HeaderMapImpl : NonCopyable {
 public:
+  HeaderMapImpl() = default;
```

Comment:
general question, here and elsewhere: IIUC, the default constructor for the header map gives 0 for both the max bytes and max count, which might be surprising to users. Wondering if it'd make sense to remove the default constructors altogether and have users always explicitly specify the max values.

(Alternatives could be adding a comment here and/or using reasonable/large default values.)

Reply:
create() a header map without the max limit is the default behavior, and is used  in a lot of places, like tests which don't really care about the max HCM limits.